### PR TITLE
Revive has_self

### LINKED ISSUE #1: Should work on methods
I know you already intend to implement that, I'm just opening an issue to track it.

### PR DESCRIPTION
**Pull request by Artoria2e5**
The commits achieve the following:
* Revive the method call support from 298b0ffa2046907eced9d15c2803be0a224cc17d. The new `func(self, ...)` approach introduced in that commit does not seem to work and is not tested.
* Revive the test case removed by the aforementioned commit. (Should fix #1 and #6, maybe also #10's question.)
* Run `rustfmt` in hope that future diffs will be reduced.
* Clarify that building of wasm is not necessary in README.

### CODE REVIEW

File: wasm/src/lib.rs [hunk 1]
```diff
@@ -249,12 +274,13 @@ pub extern "C" fn momo(code: TokenStream, _attr: TokenStream) -> TokenStream {
...
-        new_inner_item.block = Box::new(conversions.fold_block(
-            std::mem::replace(new_inner_item.block.as_mut(), parse_quote!({}))));
+        new_inner_item.block = Box::new(conversions.fold_block(std::mem::replace(
+            new_inner_item.block.as_mut(),
+            parse_quote!({}),
+        )));
         let new_inner_item = Item::Fn(new_inner_item);
         quote!(#new_item #[inline(never)] #[allow(unused_mut)] #new_inner_item)
     } else {
         code_clone
```
Comment (Artoria2e5) [inline wasm/src/lib.rs:289]: I think this branch should produce a warning at least.

Comment (Artoria2e5) [reply wasm/src/lib.rs:289]: The stuff for making warnings seems bit hard to use from inside watt. Well...

File: README.md [hunk 2]
```diff
@@ -11,23 +11,42 @@ that makes the code less readable.
...
-For now, the only place where we can put the `#[momo]` annotations
-is on plain functions.
+Conversions currently supported are `Into` (`.into()`), `AsRef`
+(`.as_ref()`), and `AsMut` (`.as_mut()`).  See `enum Conversions`
+in code.
+
+
+## Notes on watt
 
 This new updated version usesi D. Tolnay's [watt] runtime to speed
```
Comment (llogiq) [inline README.md]: I think I made a typo at "uses". Not related to this PR, but appreciated if you're up for a drive-by fix.

**Review by llogiq — APPROVED** [review]
This is great! If you want to add a drive-by typo fix, even greater, otherwise let me know & I'll merge & publish.

**Comment by llogiq** [comment]
Thank you so much! :heart:

**Pull request merged**